src/main/java/com/gotocompany/depot/maxcompute/schema/SchemaDifferenceUtils.java

```diff
@@ -0,0 +1,88 @@
+package com.gotocompany.depot.maxcompute.schema;
+
+import com.aliyun.odps.TableSchema;
+import com.aliyun.odps.type.ArrayTypeInfo;
+import com.aliyun.odps.type.StructTypeInfo;
+import com.aliyun.odps.type.TypeInfo;
+import com.gotocompany.depot.maxcompute.model.MaxComputeColumnDetail;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isPrimitiveArrayType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isPrimitiveType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isStructArrayType;
+import static com.gotocompany.depot.maxcompute.util.TypeInfoUtils.isStructType;
+
+public class SchemaDifferenceUtils {
+
+    private static final String ALTER_TABLE_QUERY_TEMPLATE = "ALTER TABLE %s.%s ADD COLUMN IF NOT EXISTS %s;";
+
+    public static List<String> getSchemaDifferenceSql(TableSchema oldSchema, TableSchema newSchema, String schemaName, String tableName) {
```

Comment:
Add java documentation that why this method do return multiple ddl sql, on what scenarios?

Also do we have equivalent implementation in the bq side and why it wasn't used here?

Reply:
BQ Client support accepting whole new TableSchema, no need for us to infer the difference individually. 
As of now, there are no such API provided by Ali's team